.github/workflows/unit_tests_with_featuretools_main_branch.yml

```diff
@@ -0,0 +1,54 @@
+name: Unit Tests - EvalML with Featuretools main branch
+on:
```

Comment:
Sorry I'm late to this PR! Awesome job @chukarsten and thanks for reviewing @gsheni ! 

Don't we need to add a workflow to woodwork and featuretools to kick off this job on commits to main?

Similar to this: https://github.com/alteryx/woodwork/blob/main/.github/workflows/kickoff_featuretools_unit_tests.yml

Reply:
Added:
https://github.com/alteryx/featuretools/pull/2072